README.md

```diff
@@ -87,13 +87,20 @@ out" (using upper-case variable names is a common way to do this).
 
 ### Comments in code
 
-Comments should explain *why* the code does what it does. *What* it
-does should ideally already be evident from the code itself. If the code
-is cryptic and can't easily be simplified, explanations might well be
-needed.
+Comments should explain *What* the code does, and *why* the code does
```

Comment:
*What* -> *what*

Reply:
Otherwise, looks good to me. Although for some classes or functions, I think it might be beneficial to add also input values, output (or return) values as well as usage.